extras/mock/preciceMocked.cpp

```diff
@@ -0,0 +1,1450 @@
+    }
+    if (solverProcessIndex < 0) {
+      throw precice::Error(precice::utils::format_or_error("solverProcessIndex must be >= 0, but is {}.", solverProcessIndex));
+    }
+    if (solverProcessIndex >= solverProcessSize) {
+      throw precice::Error(precice::utils::format_or_error("solverProcessIndex={} must be smaller than solverProcessSize={}.", solverProcessIndex, solverProcessSize));
+    }
+    seed = static_cast<uint32_t>(rank) ^ 0x9e3779b9u;
+  }
+
+  ~ParticipantImpl() = default;
+
+  // Configuration data structures
+  struct MeshInfo {
+    std::string name;
+    int         dimensions = 3;
+    bool        provided   = false;
+    bool        received   = false;
+  };
+
+  struct DataInfo {
+    std::string name;
+    std::string meshName;
+    int         dimensions = 1; // scalar by default
+    bool        isRead     = false;
+    bool        isWrite    = false;
+  };
+
+  struct ConfigData {
+    std::map<std::string, MeshInfo> meshes;
+    std::vector<DataInfo>           dataItems;
+    bool                            isImplicitCoupling = false;
+    int                             maxIterations      = 1;
+    int                             currentIteration   = 0;
+    bool                            iterationConverged = false;
+  };
+
+  enum class DataMode {
+    Random,      // Mode 1: Random data
+    Buffer,      // Mode 2: Return written data buffer (default)
+    ScaledBuffer // Mode 3: Return written data buffer scaled/multiplied
+  };
+
+  struct MockDataConfig {
+    std::string         dataName;
+    std::string         meshName;
+    DataMode            mode             = DataMode::Buffer; // Default to buffer mode
+    double              scalarMultiplier = 1.0;
+    std::vector<double> vectorMultiplier; // Empty means use scalar
+  };
+
+  struct MockConfig {
+    std::map<std::string, MockDataConfig> dataConfigs; // Key: "meshName:dataName"
+    DataMode                              defaultMode = DataMode::Buffer;
+  };
+
+  // Minimal internal state
+  std::string name;
+  std::string config;
+  int         rank            = 0;
+  int         size            = 1;
+  void       *comm            = nullptr;
+  bool        initialized     = false;
+  bool        finalized       = false;
+  bool        couplingOngoing = false;
+  double      maxTimeStep     = 0.0;
+  std::mutex  mtx;
+
+  std::vector<double> dataSeries;
+  bool                useFile     = false;
+  uint32_t            seed        = 0;
+  std::size_t         currentStep = 0;
+  double              currentTime = 0.0;
+
+  ConfigData configData;
+  bool       configParsed = false;
+
+  MockConfig mockConfig;
+  bool       mockConfigParsed = false;
+
+  // Write data buffers: Key is "meshName:dataName"
+  std::map<std::string, std::vector<double>> writeBuffers;
+
+  // std::unique_ptr<FileDoubleReader> fileReader;
+
+  void parseConfig()
```

Comment:
I wonder if we could simplify the parsing, reusing existing implementations. Maintaining this will be complicated once the config format changes again.

On the other hand, I see that this is only extracting minimal information.